src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.ReplaceTypeParameterBasedOnTypeConstraintVisitor.cs

```diff
@@ -51,9 +49,9 @@ public override ITypeSymbol VisitArrayType(IArrayTypeSymbol symbol)
                 return _compilation.CreateArrayTypeSymbol(elementType, symbol.Rank);
             }
 
-            public override ITypeSymbol VisitNamedType(INamedTypeSymbol symbol)
+            public override async ValueTask<ITypeSymbol> VisitNamedType(INamedTypeSymbol symbol)
             {
-                var arguments = symbol.TypeArguments.Select(t => t.Accept(this)).ToArray();
+                var arguments = (await Task.WhenAll(symbol.TypeArguments.Select(t => t.Accept(this).AsTask())).ConfigureAwait(false)).ToArray();
```

Comment:
this pattern seems to appear a lot.  can we extract out a helper to simplify this. i.e. `await RecurseIntArray(symbol.TypeArguments).ConfigureAwait(false)`?

Reply:
Extracted `SpecializedTasks.WhenAll` to operate on `ValueTask<T>`.